libs/datamodel/connectors/datamodel-connector/src/capabilities.rs

```diff
@@ -68,6 +68,7 @@ capabilities!(
     FullTextIndex,
     SortOrderInFullTextIndex,
     MultipleFullTextAttributesPerModel,
+    DuplicateIndexDefinitionsToSameFields,
```

Comment:
I'm wondering if we should add a capability for that — it's most likely never going to be a thing on any other connector, could it be part of the mongo connector's model validations?

Reply:
I'd don't like do validations in the datamodel connector if I don't have to :(

Reply:
I refactored those quite a bit already, the only ugly part left is ConnectorError, and my next move will be deleting that. They're just regular validations with walkers now.

Reply:
Decided in slack that yeah, this could be in the connector. Also needs @tomhoule's changes, so merging this now and refactor comes after a bit.